lib/widgets/breez_avatar_dialog.dart

```diff
@@ -11,150 +13,249 @@ import 'package:image/image.dart' as DartImage;
 import 'package:image_cropper/image_cropper.dart';
 import 'package:image_picker/image_picker.dart';
 
+import 'flushbar.dart';
+
 int scaledWidth = 200;
 var _transparentImage = DartImage.Image(scaledWidth, scaledWidth);
+enum UPLOAD_STATE { UPLOAD_IN_PROGRESS }
 
 Widget breezAvatarDialog(BuildContext context, UserProfileBloc userBloc) {
+  UPLOAD_STATE _state;
   AutoSizeGroup _autoSizeGroup = AutoSizeGroup();
   BreezUserModel _currentSettings;
+  File _pickedImage;
 
   final _nameInputController = TextEditingController();
   userBloc.userPreviewStream.listen((user) {
     _nameInputController.text = user.name;
     _currentSettings = user;
   });
 
-  Future _pickImage(BuildContext context) async {
-    return ImagePicker.pickImage(source: ImageSource.gallery).then((file) {
-      ImageCropper.cropImage(
-        sourcePath: file.path,
-        aspectRatio: CropAspectRatio(ratioX: 1.0, ratioY: 1.0),
-      ).then((file) {
-        if (file != null) {
-          file.readAsBytes().then(scaleAndFormatPNG).then((image) {
-            userBloc.uploadImageSink.add(image);
-          });
-        }
+  Future<File> _pickImage() async {
+    try {
+      File image;
+      await ImagePicker.pickImage(source: ImageSource.gallery)
+          .then((file) async {
+        await ImageCropper.cropImage(
+          sourcePath: file.path,
+          aspectRatio: CropAspectRatio(ratioX: 1.0, ratioY: 1.0),
+        ).then((file) {
+          if (file != null) {
+            image = file;
+          }
+        });
```

Comment:
The whole function can be simplified:

```suggestion
  Future<File> _pickImage() async {
  return ImagePicker.pickImage(source: ImageSource.gallery).then((file) {
    return ImageCropper.cropImage(
        sourcePath: file.path,
        aspectRatio: CropAspectRatio(ratioX: 1.0, ratioY: 1.0));
  });
}
```